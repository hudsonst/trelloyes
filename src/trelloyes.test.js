import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';
import Card from './Card';
import App from './App';
import STORE from './STORE';


describe('Trelloyes component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<App store={STORE} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders Card', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Card title='Test Title' content='Test Content' />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders List', () => {
        const div = document.createElement('div');
        ReactDOM.render(<List key='1' header='Test header' cards={['b', 'c', 'd', 'f', 'h', 'i', 'k']} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(<App store={STORE} />)
          .toJSON();
        expect(tree).toMatchSnapshot();  
        });
    
    it('renders List as expected', () => {
            const tree = renderer
              .create(<List key='1' header='Test header' cards={['b', 'c', 'd', 'f', 'h', 'i', 'k']} />)
              .toJSON();
            expect(tree).toMatchSnapshot();  
            });

    it('renders Card as expected', () => {
            const tree = renderer
              .create(<Card title='Test Title' content='Test Content' />)
              .toJSON();
            expect(tree).toMatchSnapshot();  
            });
    
});