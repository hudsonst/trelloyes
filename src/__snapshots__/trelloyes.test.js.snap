// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trelloyes component renders Card as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    Test Title
  </h3>
  <p>
    Test Content
  </p>
</div>
`;

exports[`Trelloyes component renders List as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Test header
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3 />
      <p />
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3 />
      <p />
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3 />
      <p />
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3 />
      <p />
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3 />
      <p />
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3 />
      <p />
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3 />
      <p />
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;

exports[`Trelloyes component renders the UI as expected 1`] = `
<main
  className="App"
>
  <header
    className="App-header"
  >
    <h1>
      Trelloyes!
    </h1>
  </header>
  <div
    className="App-list"
  >
    <section
      className="List"
    >
      <header
        className="List-header"
      >
        <h2>
          First list
        </h2>
      </header>
      <div
        className="List-cards"
      >
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            First card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Second card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Fifth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Sixth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Seventh card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Tenth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Twelfth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Thirteenth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <button
          className="List-add-button"
          type="button"
        >
          + Add Random Card
        </button>
      </div>
    </section>
    <section
      className="List"
    >
      <header
        className="List-header"
      >
        <h2>
          Second list
        </h2>
      </header>
      <div
        className="List-cards"
      >
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Second card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Third card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Fourth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Sixth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Eighth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Ninth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Eleventh card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <button
          className="List-add-button"
          type="button"
        >
          + Add Random Card
        </button>
      </div>
    </section>
    <section
      className="List"
    >
      <header
        className="List-header"
      >
        <h2>
          Third list
        </h2>
      </header>
      <div
        className="List-cards"
      >
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            First card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Second card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Third card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Fourth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Fifth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Sixth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Seventh card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Eighth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Ninth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Tenth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Eleventh card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Twelfth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Thirteenth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <button
          className="List-add-button"
          type="button"
        >
          + Add Random Card
        </button>
      </div>
    </section>
    <section
      className="List"
    >
      <header
        className="List-header"
      >
        <h2>
          Fourth list
        </h2>
      </header>
      <div
        className="List-cards"
      >
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Twelfth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Thirteenth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <button
          className="List-add-button"
          type="button"
        >
          + Add Random Card
        </button>
      </div>
    </section>
  </div>
</main>
`;
